src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerAssemblyLoader.Core.cs

```diff
@@ -89,19 +90,21 @@ internal DirectoryLoadContext[] GetDirectoryLoadContextsSnapshot()
             }
         }
 
-        internal void UnloadAll()
+        public void UnloadAll()
         {
-            List<DirectoryLoadContext> contexts;
+            var contexts = ArrayBuilder<DirectoryLoadContext>.GetInstance();
             lock (_guard)
             {
-                contexts = _loadContextByDirectory.Values.ToList();
+                foreach (var (_, context) in _loadContextByDirectory)
+                    contexts.Add(context);
+
                 _loadContextByDirectory.Clear();
             }
 
             foreach (var context in contexts)
-            {
                 context.Unload();
```

Comment:
Note that this can throw if the `AssemblyLoadContext` cannot be unloaded. A misbehaving analyzer could cause this to happen by stashing references to itself somewhere you did not anticipate. That is probably somewhat rare but I also think it's going to be > 0. 

This code path today doesn't really deal with exceptions because we only aggressively use it inside our unit tests. Unloading is important there because of the limitations of our test environments. 

Think we should consider how we want the IDE to behave in this case and harden this code path a bit.

Reply:
yeah, worst case, unloading doesn't work, and we have some memory growth in teh case that analyzer references are changing a lot on disk.  we can live with that.